A16-tcp-user-timeout.md

```diff
@@ -18,13 +18,18 @@ TCP_USER_TIMEOUT is a socket option that has been available on Linux since kerne
 
 ### Related Proposals:
-N/A
+[Client-side Keepalive](https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md)
+[Server-side Connection Management](https://github.com/grpc/proposal/blob/master/A9-server-side-conn-mgt.md)
 
 ## Proposal
 
-For all Linux platforms running Linux 2.6.37 and later, gRPC will set the socket option TCP_USER_TIMEOUT for TCP sockets to a default value of 20 seconds. (The system defaults might be as high as 20 minutes.)
+gRPC already has a mechanism to detect connection failures, called [keepalive](https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md). The KEEPALIVE_TIMEOUT configuration option controls the amount of time, gRPC waits for a HTTP/2 ping to be acknowledged. Since the effective aim of KEEPALIVE_TIMEOUT is similar to TCP_USER_TIMEOUT, the proposal is to reuse this configuration option to set the socket option TCP_USER_TIMEOUT. Since, TCP_USER_TIMEOUT is only available on Linux kernels 2.6.37 and later, this proposal is limited to those platforms.
+
+Reusing the KEEPALIVE_TIMEOUT configuration has a few repercussions.
+* The keepalive timeout value will also take affect on all bytes sent via TCP and not just the HTTP/2 ping.
+* TCP_USER_TIMEOUT will have the same default value and minimum value limitation as the keepalive timeout.
 
-gRPC will also provide a configuration option for both channels and servers to set the TCP_USER_TIMEOUT value. This configuration option will have millisecond accuracy and will have the same semantics as the socket option. Setting it to 0 will configure TCP_USER_TIMEOUT with the system default and not the gRPC default of 20 seconds. No minimum value to this option will be enforced by gRPC. It will be the responsibilty of the application to be prudent while setting this option. A very small TCP_USER_TIMEOUT value can affect TCP transmissions over paths with a high latency or packet loss. If the timeout occurs before the acknowledgement arrives, then the connection will be dropped.
+A very small TCP_USER_TIMEOUT value can affect TCP transmissions over paths with a high latency or packet loss. If the timeout occurs before the acknowledgement arrives, then the connection will be dropped. A very large value, on the other hand. will increase the time it takes gRPC to detect broken connections. Hence, it will be the responsibility of the application to be prudent while setting this option.
```

Comment:
"other hand. will" Period should be a comma.

Reply:
done!